packages/angular_devkit/schematics/src/engine/engine.ts

```diff
@@ -140,4 +162,42 @@ export class SchematicEngine<CollectionT extends object, SchematicT extends obje
         return hostSource;
     }
   }
+
+  createTaskExecutor(name: string): Promise<TaskExecutor> {
+    // Hard coded tasks with default factory options for now
+    switch (name) {
+      case 'npm':
```

Comment:
I think this line should be from a list of available task executors provided by the engine host.